homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,33 @@
+#include "compression.hpp"
+
+#include <algorithm>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& pgmImage) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedPGM;
+     std::for_each(pgmImage.begin(), pgmImage.end(),[&compressedPGM](auto& PixelRow){
```

Comment:
for_each for me is similar to for. You can use transform :) or std::generate (you will capture iterator and then jump between different numbers and return proper pair :) )